tests/proofs/dexter/dexter-spec.md

```diff
@@ -49,6 +49,46 @@ If any of the conditions are not satisfied, the call fails.
       orBool IsUpdating
 ```
 
+## Set Baker
+
+The contract sets its delegate to the value of `baker` (and optionally freezes the baker to that particular value) if the following conditions are satisfied:
+
+1.  the token pool is _not_ currently updating (i.e. `storage.selfIsUpdatingTokenPool = false`)
+2.  exactly 0 tez was transferred to this contract when it was invoked
+3.  the txn sender is the `storage.manager`
+4.  the baker is _not_ already frozen
+
+FIXME: This claim is incomplete because it depends on the `operations` cell which we do not have yet.
+
+```
+  claim <k> #runProof(_IsFA2, SetBaker(_Baker, FreezeBaker)) => . </k>
+        <stack> .Stack </stack>
+        <manager> Sender </manager>
+        <selfIsUpdatingTokenPool> false </selfIsUpdatingTokenPool>
+        <myamount> #Mutez(Amount) </myamount>
+        <senderaddr> Sender </senderaddr>
+        <freezeBaker> false => FreezeBaker </freezeBaker>
+        <nonce> #Nonce(N => N +Int 1) </nonce>
+    requires Amount ==Int 0
+```
+
+If any of the conditions are not satisfied, the call fails.
+
+```k
+  claim <k> #runProof(true, SetBaker(_Baker, _NewFreezeBaker)) => Aborted(?_, ?_, ?_, ?_) </k>
```

Comment:
_IsFA2 = true here, is that intentional?

Reply:
No. I'll fix that.

Reply:
Cool, if it works then go ahead and merge.

Reply:
It works. I've pushed up the fix and adding automerge.